django_sharding_library/settings_helpers.py

```diff
@@ -13,6 +15,9 @@ def database_config(environment_variable, default_database_url, shard_group=None
     db_config['TEST'] = db_config.get('TEST', {})
     db_config['SHARD_GROUP'] = shard_group
 
+    if not db_config['NAME'] and database_name is not None:
```

Comment:
Hmmm I agree that if there's a precedence it should be the name you set, rather than the one in the URL.